### PR TITLE
chore: CRW-3196 regen yarn.lock v1

### PR DESCRIPTION
**Pull request by nickboldt**
run 'rm...

  ### What does this PR do?

chore: [CRW-3196](https://issues.redhat.com//browse/CRW-3196) regen yarn.lock v1

run 'rm -f yarn.lock tools/build/yarn.lock; yarn install --mode update-lockfile; cp yarn.lock tools/build/yarn.lock';
install workspace-tools plugin, and verify if we're building here with yarn v1 or 2+ (workspace command seems to be 2+);
run the build from the correct folder when doing pr ts check (thanks @svor );
Valerii says we can remove yarn 3 binary; switch to yarn v1.22.19;
add GH action to regen yarn.lock and copy to tools/build/ folder

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
N/A (or see commit message above for issue number)

### How to test this PR?
N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by nickboldt** [comment]
We might be able to also remove https://github.com/eclipse-che/che-plugin-registry/blob/main/.github/workflows/dependabot-pr-workaround.yml as this switches us to Yarn v1, which isn't affected by the dependabot issue linked in the GH action (https://github.com/dependabot/dependabot-core/issues/1297) ... which is also resolved by now.

**Comment by svor** [comment]
it seems that something is running recursively :)

**Review by svor — CHANGES_REQUESTED** [review]
Blocked it because PR checks are running recursively
(I have a lot of notifications  :flushed:)

File: .github/workflows/regen-yarn-lock-and-copy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+          token: ${{secrets.CHE_BOT_GITHUB_TOKEN}}
+      - name: Copy yarn.lock
+        run: |
+          cp -f yarn.lock tools/build/yarn.lock
```
Comment (svor) [inline .github/workflows/regen-yarn-lock-and-copy.yml:34]: probably we have to check if tools/build/yarn.lock exists and remove it before copying

**Comment by svor** [comment]
> We might be able to also remove https://github.com/eclipse-che/che-plugin-registry/blob/main/.github/workflows/dependabot-pr-workaround.yml as this switches us to Yarn v1, which isn't affected by the dependabot issue linked in the GH action ([dependabot/dependabot-core#1297](https://github.com/dependabot/dependabot-core/issues/1297)) ... which is also resolved by now.

yes, let's remove it since we jumped back to yarn 1

**Review by svor — APPROVED** [review]

**Pull request merged**